### ISSUE TITLE
go get -u github.com/go-gl/glfw/v3.2/glfw fails on OSX - Xcode 7.0.1

### ISSUE THREAD

**Issue by phbcanada** [go-gl/glfw#184]
```
go get -u github.com/go-gl/glfw/v3.2/glfw
# github.com/go-gl/glfw/v3.2/glfw
In file included from Go/src/github.com/go-gl/glfw/v3.2/glfw/c_glfw.go:4:
In file included from ./glfw/src/context.c:28:
In file included from ./glfw/src/internal.h:165:
./glfw/src/cocoa_platform.h:31:10: fatal error: 'dlfcn.h' file not found
#include <dlfcn.h>
         ^
1 error generated.
```

Which version of Xcode is required?

**Comment by dmitshur** [comment]
I have Xcode Version 8.2.1.

What version of macOS are you using?

What version of Go?

**Comment by phbcanada** [comment]
OSX 10.11.6
GO 1.8

**Comment by phbcanada** [comment]
I've just updated my Xcode to 8.2.1 and am still getting the same error.

**Comment by dmitshur** [comment]
Strange, it should work.

Does the file `dlfcn.h` exist on your computer? In what locations?

This is what I get:

```
$ sudo find / -name 'dlfcn.h'
/Applications/Xcode.app/Contents/Developer/Platforms/AppleTVOS.platform/Developer/SDKs/AppleTVOS.sdk/usr/include/dlfcn.h
/Applications/Xcode.app/Contents/Developer/Platforms/AppleTVSimulator.platform/Developer/SDKs/AppleTVSimulator.sdk/usr/include/dlfcn.h
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS.sdk/usr/include/dlfcn.h
/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator.sdk/usr/include/dlfcn.h
/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/dlfcn.h
/Applications/Xcode.app/Contents/Developer/Platforms/WatchOS.platform/Developer/SDKs/WatchOS.sdk/usr/include/dlfcn.h
/Applications/Xcode.app/Contents/Developer/Platforms/WatchSimulator.platform/Developer/SDKs/WatchSimulator.sdk/usr/include/dlfcn.h
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift-migrator/sdks/AppleTVOS.sdk/usr/include/dlfcn.h
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift-migrator/sdks/AppleTVSimulator.sdk/usr/include/dlfcn.h
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift-migrator/sdks/iPhoneOS.sdk/usr/include/dlfcn.h
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift-migrator/sdks/iPhoneSimulator.sdk/usr/include/dlfcn.h
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift-migrator/sdks/MacOSX.sdk/usr/include/dlfcn.h
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift-migrator/sdks/WatchOS.sdk/usr/include/dlfcn.h
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift-migrator/sdks/WatchSimulator.sdk/usr/include/dlfcn.h
```

A couple ideas for things you can try. I'm not sure what's required to make it work:

- Install latest version of "Command Line Tools for Xcode" via `xcode-select --install`.
- Update to macOS 10.12.3. That's the version I have. But I'm pretty sure I didn't have this issue even when I was on 10.11.6.

**Comment by phbcanada** [comment]
Yes, I have it in the same places.
Very strange that it doesn't find it.  :(

**Comment by dmitshur** [comment]
What's the output if you do the following:

```bash
$ echo '#include <dlfcn.h>

int main() {
    return 0;
}' > main.cpp
$ clang++ -v main.cpp
```

I get something that mentions:

```
#include "..." search starts here:
#include <...> search starts here:
 /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1
 /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../lib/clang/8.0.0/include
 /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/include
 /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk/usr/include
 /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk/System/Library/Frameworks (framework directory)
End of search list.
```

What about you?

**Comment by phbcanada** [comment]
Well after lots of digging and experimenting I re-installed command line tools from: https://developer.apple.com/download/more/ which recreated the missing /usr/include directory. Now it finds the includes but I get several thousand other framework related errors. It seems there's must be something else broken in my Xcode installation.

**Comment by dmitshur** [comment]
Got it.

It sounds like it's a local setup issue, not an issue in go-gl/glfw.

Apple has released macOS 10.12.4 update today, as well as Xcode 8.3. Perhaps updating will fix your issues. If not, you might need to take more drastic actions, such as reinstalling macOS or restoring from a working backup.
